core/templates/domain/exploration/StatesObjectFactory.ts

```diff
@@ -120,22 +126,28 @@ export class States {
     return finalStateNames;
   }
 
-  getAllVoiceoverLanguageCodes(): string[] {
-    var allAudioLanguageCodes = [];
-    for (var stateName in this._states) {
-      var state = this._states[stateName];
-      var contentIdsList = state.recordedVoiceovers.getAllContentId();
-      contentIdsList.forEach(function(contentId) {
-        var audioLanguageCodes = (
-          state.recordedVoiceovers.getVoiceoverLanguageCodes(contentId));
-        audioLanguageCodes.forEach(function(languageCode) {
-          if (allAudioLanguageCodes.indexOf(languageCode) === -1) {
-            allAudioLanguageCodes.push(languageCode);
-          }
-        });
+  _getAllLanguageCodesFor(
+      translationType: 'recordedVoiceovers' | 'writtenTranslations'
```

Comment:
Make the two strings into (local) named constants, and use those named constants throughout the file.